modules/miner/mine.go

```diff
@@ -94,18 +105,22 @@ func (m *Miner) solveBlock(blockForWork types.Block, target types.Target, iterat
 	// solveBlock could operate on a pointer, but it's not strictly necessary
 	// and it makes calling weirder/more opaque.
 	b = blockForWork
+	bRoot := b.MerkleRoot()
 
 	// Iterate through a bunch of nonces (from a random starting point) and try
 	// to find a winnning solution.
 	for maxNonce := b.Nonce + iterations; b.Nonce != maxNonce; b.Nonce++ {
-		if b.CheckTarget(target) {
+		if fastCheckTarget(target, b, bRoot) {
 			err = m.state.AcceptBlock(b)
-			if build.DEBUG {
-				if err != nil {
-					println(err.Error())
-				}
+			if err != nil {
+				print("mined a block, but with an error: ")
+				println(err.Error())
+				m.tpool.PurgeTransactionPool()
 			}
 			solved = true
+			if build.Release != "testing" {
+				println("Found a block! If the block is not orphaned, you will receive the reward after 50 more blocks have been mined. Blocks are only orphaned when two miners find a block at the same time, as only 1 block can be accepted.")
+			}
```

Comment:
how about just `println("Found a block! You will receive the reward in 50 blocks")`